train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java

```diff
@@ -51,4 +53,15 @@ public void emergencyBreak() {
 		this.referenceSpeed = 0;
 	}
 
+	@Override
```

Comment:
Good job! For a more effective simulation I would set the interval to a lower value.